### PR TITLE
Test suite autodetect datastream.

### PR DESCRIPTION
**Pull request by matejak**
The frequent use case is that a single datastream is built in the build directory, and one wants it to be tested by the test suite.
This change add autodetection capabilities, so the --datastream option can be left out. If it is left out and there is not a single datastream in the build dir, an error is displayed instead.

As a side-effect, the `--datastream` option won't be required any longer, which is [generally recommended](https://docs.python.org/3/library/argparse.html#required).

> Required options are generally considered bad form because users expect options to be optional, and thus they should be avoided when possible.

### CODE REVIEW

File: tests/test_suite.py [hunk 1]
```diff
@@ -216,10 +216,32 @@ def auto_select_xccdf_id(datastream, bench_number):
...
+    datastreams = glob(os.path.sep.join(build_dir_path + [ds_glob]))
+    return datastreams
+
+
+def get_unique_datastream():
+    datastreams = get_datastreams()
+    if len(datastreams) == 1:
+        return datastreams[0]
+    msg = ("Unable to auto-detect the datastream, "
+           "expected to find a single one in the build dir, but")
```
Comment (yuumasato) [inline tests/test_suite.py]: I wouldn't say that it is expected that there is only single datastream in build dir, I have othen more than one.

File: tests/test_suite.py [hunk 2]
```diff
@@ -216,10 +216,32 @@ def auto_select_xccdf_id(datastream, bench_number):
...
+    if not datastreams:
+        raise RuntimeError(msg + " found none.")
+    raise RuntimeError(
+        msg + " found multiple ones: {0}".format(datastreams))
```
Comment (yuumasato) [inline tests/test_suite.py]: If multiple DataStreams are found, the tool should recommend use of `--datastream` option.

**Comment by matejak** [comment]
Thanks for the feedback, comments are addressed.

File: tests/test_suite.py [hunk 3]
```diff
@@ -218,21 +218,23 @@ def auto_select_xccdf_id(datastream, bench_number):
...
-    msg = ("Unable to auto-detect the datastream, "
-           "expected to find a single one in the build dir, but")
+    msg = ("Autodetection of the datastream file is possible only when there is "
+           "a single one in the build dir, but")
     if not datastreams:
-        raise RuntimeError(msg + " found none.")
+        raise RuntimeError(msg + " there is none.")
     raise RuntimeError(
-        msg + " found multiple ones: {0}".format(datastreams))
+        msg + " there are {0} of them. Use the --datastream option to select "
```
Comment (yuumasato) [inline tests/test_suite.py]: The line can get rather long:
```shell
Setting console output to log level INFO
ERROR - Error occurred during options normalization: Autodetection of the datastream file is possible only when there is a single one in the build dir, but there are 28 of them. Use the --datastream option to select e.g. ['../build/ssg-debian8-ds.xml', '../build/ssg-rhel7-ds.xml', '../build/ssg-wrlinux8-ds.xml', '../build/ssg-wrlinux1019-ds.xml', '../build/ssg-ubuntu1404-ds.xml', '../build/ssg-ol8-ds.xml', '../build/ssg-centos6-ds.xml', '../build/ssg-fuse6-ds.xml', '../build/ssg-centos7-ds.xml', '../build/ssg-eap6-ds.xml', '../build/ssg-ubuntu1604-ds.xml', '../build/ssg-ubuntu1804-ds.xml', '../build/ssg-rhv4-ds.xml', '../build/ssg-rhosp13-ds.xml', '../build/ssg-chromium-ds.xml', '../build/ssg-fedora-ds.xml', '../build/ssg-ol7-ds.xml', '../build/ssg-sl6-ds.xml', '../build/ssg-rhel8-ds.xml', '../build/ssg-sle12-ds.xml', '../build/ssg-firefox-ds.xml', '../build/ssg-sl7-ds.xml', '../build/ssg-sle11-ds.xml', '../build/ssg-rhel6-ds.xml', '../build/ssg-opensuse-ds.xml', '../build/ssg-jre-ds.xml', '../build/ssg-ocp3-ds.xml', '../build/ssg-centos8-ds.xml']
```
And in the terminal it actually wraps up.
I like that they short and easy to copy-paste when adding `--datastream` option.

**Pull request merged**